packages/map-toolkit/src/hover-coordinate/use-hover-coordinate.ts

```diff
@@ -0,0 +1,121 @@
+import { coordinateSystems, createCoordinate } from '@accelint/geo';
+import { useContext, useEffect, useState } from 'react';
+import { MapEvents } from '../deckgl/base-map/events';
+import { MapContext } from '../deckgl/base-map/provider';
+import type { UniqueId } from '@accelint/core';
+import type { MapHoverEvent } from '../deckgl/base-map/types';
+
+/**
+ * Supported coordinate format types for displaying map coordinates.
+ *
+ * @typedef {'dd' | 'ddm' | 'dms' | 'mgrs' | 'utm'} FormatTypes
+ * @property dd - Decimal Degrees (e.g., "45.50000000 E / 30.25000000 N")
+ * @property ddm - Degrees Decimal Minutes (e.g., "45 30 E / 30 15 N")
+ * @property dms - Degrees Minutes Seconds (e.g., "45 30 0 E / 30 15 0 N")
+ * @property mgrs - Military Grid Reference System
+ * @property utm - Universal Transverse Mercator
+ */
+export type FormatTypes = 'dd' | 'ddm' | 'dms' | 'mgrs' | 'utm';
+
+const MAX_LONGITUDE = 180;
+const LONGITUDE_RANGE = 360;
+const COORDINATE_PRECISION = 8;
+const prepareCoord = (coord: [number, number]) => {
+  // Normalize longitude to -180 to 180 range (handles International Date Line)
+  let lon = coord[0];
+  if (lon > MAX_LONGITUDE) {
+    lon -= LONGITUDE_RANGE;
+  } else if (lon < -MAX_LONGITUDE) {
+    lon += LONGITUDE_RANGE;
+  }
+
+  const lat = coord[1];
+  const lonStr = `${Math.abs(lon).toFixed(COORDINATE_PRECISION)} ${lon < 0 ? 'W' : 'E'}`;
+  const latStr = `${Math.abs(lat).toFixed(COORDINATE_PRECISION)} ${lat < 0 ? 'S' : 'N'}`;
+
+  return `${lonStr} / ${latStr}`;
+};
+
+/**
+ * React hook that tracks and formats the mouse hover position coordinates on a map.
+ * Listens to map hover events via the event bus and converts coordinates to various formats.
+ *
+ * @param id - Optional map instance ID. If not provided, attempts to use MapProvider context.
+ * @returns Object containing the formatted coordinate string and format setter function
+ * @throws {Error} When no id is provided and hook is used outside MapProvider context
+ *
+ * @example
+ * ```tsx
+ * const MAP_ID = uuid();
+ *
+ * function CoordinateDisplay() {
+ *   const { formattedCoord, setFormat } = useHoverCoordinate(MAP_ID);
+ *
+ *   return (
+ *     <div>
+ *       <select onChange={(e) => setFormat(e.target.value as FormatTypes)}>
+ *         <option value="dd">Decimal Degrees</option>
+ *         <option value="mgrs">MGRS</option>
+ *       </select>
+ *       <div>{formattedCoord}</div>
+ *     </div>
+ *   );
+ * }
+ * ```
+ */
+export function useHoverCoordinate(id?: UniqueId) {
+  const contextId = useContext(MapContext);
+  const actualId = id ?? contextId;
+
+  if (!actualId) {
+    throw new Error(
+      'useHoverCoordinate requires either an id parameter or to be used within a MapProvider',
+    );
+  }
+
+  const [formattedCoord, setFormattedCoord] = useState('--, --');
+  const [format, setFormat] = useState<FormatTypes>('dd');
+  const create = createCoordinate(coordinateSystems.dd, 'LONLAT');
```

Comment:
this could be moved outside the hook since there are no internal deps. That would prevent new ones being created every render.